src/Arcus.Observability.Telemetry.Serilog.Enrichers/Extensions/LoggerEnrichmentConfigurationExtensions.cs

```diff
@@ -44,5 +48,52 @@ public static LoggerConfiguration WithKubernetesInfo(this LoggerEnrichmentConfig
+        /// Adds the <see cref="CorrelationInfoEnricher{TCorrelationInfo}"/> to the logger enrichment configuration which adds the <see cref="CorrelationInfo"/> information from the current context.
+        /// </summary>
+        /// <param name="enrichmentConfiguration">The configuration to add the enricher.</param>
+        /// <param name="correlationInfoAccessor">The accessor implementation for the <see cref="CorrelationInfo"/> model.</param>
+        public static LoggerConfiguration WithCorrelationInfo(this LoggerEnrichmentConfiguration enrichmentConfiguration, ICorrelationInfoAccessor correlationInfoAccessor)
+        {
+            Guard.NotNull(enrichmentConfiguration, nameof(enrichmentConfiguration));
+            Guard.NotNull(correlationInfoAccessor, nameof(correlationInfoAccessor));
+
+            return WithCorrelationInfo<CorrelationInfo>(enrichmentConfiguration, correlationInfoAccessor);
+        }
+
+        /// <summary>
+        /// Adds the <see cref="CorrelationInfoEnricher{TCorrelationInfo}"/> to the logger enrichment configuration which adds the custom <see cref="CorrelationInfo"/> information from the current context.
+        /// </summary>
+        /// <typeparam name="TCorrelationInfo">The type of the custom <see cref="CorrelationInfo"/> model.</typeparam>
+        /// <param name="enrichmentConfiguration">The configuration to add the enricher.</param>
+        /// <param name="correlationInfoAccessor">The accessor implementation for the <see cref="CorrelationInfo"/> model.</param>
+        public static LoggerConfiguration WithCorrelationInfo<TCorrelationInfo>(
+            this LoggerEnrichmentConfiguration enrichmentConfiguration, 
+            ICorrelationInfoAccessor<TCorrelationInfo> correlationInfoAccessor) where TCorrelationInfo : CorrelationInfo
+        {
+            Guard.NotNull(enrichmentConfiguration, nameof(enrichmentConfiguration));
+            Guard.NotNull(correlationInfoAccessor, nameof(correlationInfoAccessor));
+
+            return WithCorrelationInfo(enrichmentConfiguration, correlationInfoAccessor, enrichAdditionalCorrelationInfo: null);
+        }
+
+        /// <summary>
+        /// Adds the <see cref="CorrelationInfoEnricher{TCorrelationInfo}"/> to the logger enrichment configuration which adds the custom <see cref="CorrelationInfo"/> information from the current context.
+        /// </summary>
+        /// <typeparam name="TCorrelationInfo">The type of the custom <see cref="CorrelationInfo"/> model.</typeparam>
+        /// <param name="enrichmentConfiguration">The configuration to add the enricher.</param>
+        /// <param name="correlationInfoAccessor">The accessor implementation for the <see cref="CorrelationInfo"/> model.</param>
+        /// <param name="enrichAdditionalCorrelationInfo">The function to enrich additional information from the custom <see cref="CorrelationInfo"/> model</param>
+        public static LoggerConfiguration WithCorrelationInfo<TCorrelationInfo>(
```

Comment:
I'd not use  this enrich action, has to be done by providing a different enricher instead of the default one